Property 'code' is missing in type '{ message: string; }' but required in type 'RollupWarning'.

The TypeScript declaration for `RollupWarning` contains a `code` property that should be optional, but isn't.

<!--
  ⚡️ katchow! We 💛 issues.

  Please - do not - remove this template.
  Please - do not - skip or remove parts of this template.
  Or your issue may be closed.

  👉🏽 Need help or tech support? Please don't open an issue!
  Head to https://gitter.im/rollup/rollup or https://stackoverflow.com/questions/tagged/rollupjs

  ❤️ Rollup? Please consider supporting our collective:
  👉 https://opencollective.com/rollup/donate
-->

- Rollup Version: 1.0.0
- Operating System (or Browser): 
- Node Version: 11.2.0

### How Do We Reproduce?

<!--
  Issues without minimal reproductions will be closed! Please provide one by:
  1. Using the REPL at https://rollupjs.org/repl, or
  2. Work to isolate the problem and provide the exact steps in this issue, or
  3. Provide a minimal repository link (Read https://git.io/fNzHA for instructions).
     These may take more time to triage than the other options.
-->

Start developing a plugin, using TypeScript.

**package.json**

```json
{
    "name": "rollup-plugin-hmm",
    "devDependencies": {
        "rollup": "^1.0.0",
        "typescript": "^3.2.2"
    },
    "scripts": {
        "start": "tsc index.ts"
    }
}

```

**index.ts**

```js
import { Plugin } from 'rollup';

export default function(): Plugin {
    return {
        name: 'hmm',
        load(id) {
            this.warn({ message: 'hmm' });
            return null;
        }
    };
}
```

Now do:
```sh
$ npm start
```

### Expected Behavior
No errors

### Actual Behavior
```
> tsc index.ts

index.ts:7:23 - error TS2345: Argument of type '{ message: string; }' is not assignable to parameter of type 'string | RollupWarning'.
  Property 'code' is missing in type '{ message: string; }' but required in type 'RollupWarning'.

7             this.warn({ message: 'hmm' });
                        ~~~~~~~~~~~~~~~~~~

  node_modules/rollup/dist/rollup.d.ts:15:2
    15  code: string;
        ~~~~
    'code' is declared here.


Found 1 error.
```


<!--
  Most issues can be expressed or demonstrated through the REPL or a repository.
  However, the situation may arise where some small code snippets also need to
  be provided. In that situation, please add your code below using
  Fenced Code Blocks (https://help.github.com/articles/creating-and-highlighting-code-blocks/)
-->

--- a/src/rollup/types.d.ts
+++ b/src/rollup/types.d.ts
@@ -12,7 +12,6 @@ export interface RollupError extends RollupLogProps {
 }
 
 export interface RollupWarning extends RollupLogProps {
-	code: string;
 	exporter?: string;
 	exportName?: string;
 	guess?: string;

